dbt_meshify/main.py

```diff
@@ -192,20 +255,29 @@ def add_contract(select, exclude, project_path, selector, read_catalog, public_o
     logger.info(f"Selected {len(resources)} resources: {resources}")
     models = filter(lambda x: x.startswith("model"), resources)
     if public_only:
-        models = filter(lambda x: project.get_manifest_node(x).access == "public", models)
-    logger.info(f"Adding contracts to models in selected resources...")
-    for model_unique_id in models:
-        model_node = project.get_manifest_node(model_unique_id)
-        model_catalog = project.get_catalog_entry(model_unique_id)
-        meshify_constructor = DbtMeshConstructor(
-            project_path=project_path, node=model_node, catalog=model_catalog
+        models = filter(
```

Comment:
this could be a good opportunity for us to refactor this so that we don't need the `public_only` arg at all. we only use this for the group command, and starting with dbt 1.6+ we can just use `--select access:pubic` instead on this lambda

Reply:
This makes sense! Given that this would be removing a command line argument, I believe we should tackle that in a separate issue/PR. It's not a "**breaking**" breaking change, but it is changing UX a bit.